jekyll/_cci2/vs-code-extension-overview.adoc

```diff
@@ -165,12 +165,51 @@ Both of these commands can be run by:
 
 [#open-source-language-server]
 === Open source language server
 
-The config helper is based on a dedicated language server specific for CircleCI YAML files, which is open source. You can view its source code, contribute and add issues directly on the project repository: link:https://github.com/CircleCI-Public/circleci-yaml-language-server[circleci-yaml-language-server].
+The config helper is based on a dedicated language server specific for CircleCI YAML files, which is open source. You can view its source code, contribute and add issues directly on the project repository: link:https://github.com/CircleCI-Public/circleci-yaml-language-server[CircleCI YAML language server].
 
 You can also integrate the language server into any editor which supports the Language Server Protocol, and build your own plugin to benefit from config helper capabilities in your favourite editor.
 
+[#test-run-your-config-from-vs-code]
+== Test run your config from VS Code
+
+NOTE: The ability to trigger a pipeline from VS Code with a _custom_ config is currently in **closed preview**. This feature is available on the **Free Plan** for accounts integrated through the **GitHub OAuth app**, or **Bitbucket**. To find out which GitHub account type you have, see the xref:github-integration#[GitHub OAuth app integration] page.
+
+Trigger pipelines from VS Code to iterate on your CircleCI config without committing your trial and error changes to your version control system. Run and validate your full pipeline, or select jobs and workflows to validate individually. View the results of your test runs in the CircleCI web app, just the same as any other pipeline.
+
+[#prerequisites]
+=== Prerequisites
+
+* v2.0 or higher of the CircleCI VS Code extension
+* A CircleCI account integrated through the GitHub OAuth app, or Bitbucket. To find out which GitHub account type you have, see the xref:github-integration#[GitHub OAuth app integration] page
+* You must be on the link:https://circleci.com/pricing/[Free plan]
+* You need to opt-in to this feature per project. For your project, navigate to menu:Project Settings[VS Code integration] and toggle the "Run local config from VS Code" to **On**.
+* Your project must **not** make use of xref:dynamic-config#[dynamic configuration]. This feature is disabled for projects that use dynamic configuration.
+
+[#steps]
+=== Steps
+
+[WARNING]
+====
+Triggering a pipeline with a _custom_ configuration can have side effects, for example, unwanted deployment. We recommend using this feature **only** on feature branches.
+
+Test run pipelines triggered from VS Code have access to contexts, and project environment variables, based on which branch they are run on, and user permissions.
+====
+
+The following steps show how to trigger a test run pipeline from the VS Code extension:
```

Comment:
Genuine question - does it sound better to go for this, using "test run" as a verb?
> The following steps show how to **test run a pipeline**

Reply:
Yes!